x-pack/metricbeat/module/meraki/device_health/device_network_health_channel_utilization.go

```diff
@@ -0,0 +1,96 @@
+
+	for _, network := range *networks {
+		for _, product_type := range network.ProductTypes {
+			if product_type == "wireless" {
+				networkHealthUtilization, res, err := client.Networks.GetNetworkNetworkHealthChannelUtilization(network.ID, &meraki_api.GetNetworkNetworkHealthChannelUtilizationQueryParams{})
+				if err != nil {
+					//"This endpoint is only available for networks on MR 27.0 or above."
+					// We just ignore this error but do not append to the list
+					if !(strings.Contains(string(res.Body()), "MR 27.0")) {
+						//Any other problem we are going to return an error
+						return nil, fmt.Errorf("Networks.GetNetworkNetworkHealthChannelUtilization failed; [%d] %s. %w", res.StatusCode(), res.Body(), err)
+					}
+				} else {
+					//fmt.Printf("\n#Added Networks.GetNetworkNetworkHealthChannelUtilization networkInfo; [%d] %s.\n###################", res.StatusCode(), res.Body())
+					networkHealthUtilizations = append(networkHealthUtilizations, networkHealthUtilization)
+				}
+
+			}
+		}
+	}
+	return networkHealthUtilizations, nil
+}
+
+func reportNetworkHealthChannelUtilization(reporter mb.ReporterV2, organizationID string, devices map[Serial]*Device, networkHealthUtilizations []*meraki_api.ResponseNetworksGetNetworkNetworkHealthChannelUtilization) {
+	metrics := []mapstr.M{}
+	for _, networkHealthUtil := range networkHealthUtilizations {
+		for _, network := range *networkHealthUtil {
+
+			wifi0_encountered := false
+			wifi1_encountered := false
+
+			metric := mapstr.M{
+				"network.health.channel.radio.serial": network.Serial,
+				"network.health.channel.radio.model":  network.Model,
+				"network.health.channel.radio.tags":   network.Tags,
+			}
+
+			if device, ok := devices[Serial(network.Serial)]; ok {
+				metric["device.address"] = device.Address
+				metric["device.firmware"] = device.Firmware
+				metric["device.imei"] = device.Imei
+				metric["device.lan_ip"] = device.LanIP
+				metric["device.location"] = device.Location
+				metric["device.mac"] = device.Mac
+				metric["device.model"] = device.Model
+				metric["device.name"] = device.Name
+				metric["device.network_id"] = device.NetworkID
+				metric["device.notes"] = device.Notes
+				metric["device.product_type"] = device.ProductType
+				metric["device.serial"] = device.Serial
+				metric["device.tags"] = device.Tags
+
+			}
+
+			for _, wifi0 := range *network.Wifi0 {
+				metric["network.health.channel.radio.wifi0.start_time"] = wifi0.StartTime
+				metric["network.health.channel.radio.wifi0.end_time"] = wifi0.EndTime
+				metric["network.health.channel.radio.wifi0.utilization80211"] = wifi0.Utilization80211
+				metric["network.health.channel.radio.wifi0.utilizationNon80211"] = wifi0.UtilizationNon80211
+				metric["network.health.channel.radio.wifi0.utilizationTotal"] = wifi0.UtilizationTotal
+				wifi0_encountered = true
+				metrics = append(metrics, metric)
```

Comment:
are these `append` calls here and line 86 supposed to be here as well as the call on line 90?

Reply:
oh i see it's if neither of these blocks was entered

Reply:
if we don't have utilization metrics (i.e. neither of these blocks were entered), why bother reporting any metric events at all?

Reply:
If you look at lines 46 to 60, it is possible they returned some network health, but the loops were nil.  That was my logic, there is very little data though, if the for loops do not have data. I actually used this in a few locations, where there is sometimes a little or lot of data before a looping structure.

Reply:
There are some case where is probably warranted due to several values being returned, but perhaps the for loop is nil ... if you search on _encountered you can see a few other spots.